### PR TITLE
[ROCm] Update clock intrinsic handling for AMD gfx11 family

### PR DESCRIPTION
**Pull request by pruthvistony**
cc @jeffdaily @sunway513 @jithunnair-amd @ROCmSupport

### CODE REVIEW

**Comment by pruthvistony** [comment]
The CUDA timeout - https://hud.pytorch.org/pr/97005#12073397008
seems a different issue not related to this PR.

**Review by jithunnair-amd — APPROVED** [review]

File: aten/src/ATen/cuda/Sleep.cu [hunk 1]
```diff
@@ -7,12 +7,21 @@ namespace at {
 namespace cuda {
 namespace {
 __global__ void spin_kernel(int64_t cycles) {
+  // Few AMD specific GPUs have different clock intrinsic
+#ifdef __GFX11__
```
Comment (malfet) [inline aten/src/ATen/cuda/Sleep.cu:11]: Would be nice to make it a bit more explicit, that it does not apply to CUDA
```suggestion
#if defined(__GFX11__) && !defined(__CUDA_ARCH__)
```

**Review by malfet — APPROVED** [review]

File: aten/src/ATen/cuda/Sleep.cu (hunk 1 above)
Comment (pruthvistony) [reply aten/src/ATen/cuda/Sleep.cu:11]: @malfet ,
I can also make it ROCm specific adding `defined(USE_ROCM)`

Comment (pruthvistony) [reply aten/src/ATen/cuda/Sleep.cu]: Updated the code. Will wait for CI to finish before merging.

File: aten/src/ATen/cuda/Sleep.cu [hunk 2]
```diff
@@ -7,12 +7,21 @@ namespace at {
 namespace cuda {
 namespace {
 __global__ void spin_kernel(int64_t cycles) {
+  // Few AMD specific GPUs have different clock intrinsic
+#if defined(__GFX11__) && defined(USE_ROCM) && !defined(__CUDA_ARCH__)
```
Comment (jithunnair-amd) [inline aten/src/ATen/cuda/Sleep.cu:11]: Do we need both `USE_ROCM` and `__CUDA_ARCH__`?

Comment (pruthvistony) [reply aten/src/ATen/cuda/Sleep.cu:11]: To make the sure the code is enabled only for ROCm.

**Pull request closed without merging**